extensions/transform/org.eclipse.smarthome.transform.exec/README.md

```diff
@@ -6,14 +6,45 @@ The external program must either be in the executable search path of the server
 
 ## Example
 
-With the command line 
+**.items**
 
+This will replace the visible label in the UI with the transformation you apply with the command <YourCommand>.
+```java
+String yourItem "Some info  [EXEC(/absolute/path/to/your/TransformProgram %s):]"
 ```
+  
+**.thing**
+
+This example shows the transformation of the input channel of the Exec Binding with "%2s" and the transformation of the return value of the execution, the output channel, with "%s".
+```java
+Thing exec:command:yourCommand [ command="/absolute/path/to/your/ExecProgram %2s" , transform="EXEC(/absolute/path/to/your/TransformProgram %s)"]
+```
+
+**.rules**
+
+```java
+rule "Your Rule Name"
+  when
+    Item YourTriggeringItem changed
+then
+    var formatted = transform("EXEC","/absolute/path/to/your/TransformProgram ", YourTriggeringItem.state.toString)
+    yourFormattedItem.sendCommand(formatted.toString) 
+end
+```
+
+**Example with a program**
+
+Substitut the `/absolute/path/to/your/TransformProgram` with
+```java
 /bin/date -v1d -v+1m -v-1d -v-%s
 ```
 
-return a string showing the last weekday of the month.
+The execution returns a string showing the last weekday of the month.
 
-| input | output |
-|-------|--------|
+| input | output                         |
+|-------|--------------------------------|
 | `fri` | `Fri 31 Mar 2017 13:58:47 IST` |
+
+## Source
```

Comment:
Please don't add any references to openHAB here in Eclipse SmartHome. While openHAB is _based_ upon Eclipse SmartHome, it is _just one_ of the existing solutions.